helpers/localize.js

```diff
@@ -0,0 +1,40 @@
+import { html } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+
+export const acceptedTags = Object.freeze(['d2l-link', 'd2l-tooltip-help', 'p', 'br', 'b', 'strong', 'i', 'em']);
+
+const markupError = `localize() rich-text replacements must use markup templates with only the following accepted elements: ${acceptedTags}. [link to docs]`;
+const regex = new RegExp(`<(?!\/?(${acceptedTags.join('|')}))`);
+
+export function validateMarkup(parts) {
+	return parts.map(p => {
+		debugger;
+		if (p._markup) return p;
+		else if (p.constructor !== String) throw markupError;
+		return p;
+	});
+}
+
+export function markup(str, ...parts) {
+	let reject = str.some(str => regex.test(str));
+	reject = reject || parts.some(subparts => { // subparts === chunks?
+		if (subparts.constructor === String) {
+			return regex.test(subparts);
+		}
+		else if (Array.isArray(subparts)) {
+			 return !subparts.some(subpart => subpart.constructor === String || subpart._markup);
```

Comment:
These arrays (in the `html` literal) can be infinitely nested, so some recursion is probably necessary. I've got a suggestion I can put up soon (let's see if we can get infinite recursion on PRs! 😉)

Reply:
Ah, I see. I think we are thinking of two different approaches. Mine (obviously not properly implemented here), would be to disallow even strings, so every level would be independently calling `markup` and validating. This is more like what lit itself does, but I could be convinced to allow strings and just do the recursion inside the validation logic.

Reply:
And now that I'm thinking it through, I guess lit is more forgiving. Writing that logic might be overcomplicating things, so now I'm leaning toward the recursion.

Reply:
Yeah I guess what I'm saying is this should probably be valid (it would be valid in an html literal): `markup`${['a', ['b', ['c', 'd']]]}`` - and you probably want to do all those checks all the way down.

Reply:
Okay - a draft with some additional suggestions:
https://github.com/BrightspaceUI/core/pull/3381